gix/src/repository/remote.rs

```diff
@@ -75,45 +77,46 @@ impl crate::Repository {
         self.try_find_remote_inner(name_or_url, false)
     }
 
+    fn config_spec<T: config::tree::keys::Validate>(
+        specs: Vec<std::borrow::Cow<'_, BStr>>,
+        name_or_url: &BStr,
+        key: &'static config::tree::keys::Any<T>,
+        op: gix_refspec::parse::Operation,
+    ) -> Result<Vec<gix_refspec::RefSpec>, find::Error> {
+        let kind = key.name;
+        specs
+            .into_iter()
+            .map(|spec| {
+                key.try_into_refspec(spec, op).map_err(|err| find::Error::RefSpec {
+                    remote_name: name_or_url.into(),
+                    kind,
+                    source: err,
+                })
+            })
+            .collect::<Result<Vec<_>, _>>()
+            .map(|mut specs| {
+                specs.sort();
+                specs.dedup();
+                specs
+            })
+    }
+
+    #[momo]
+    #[allow(clippy::needless_lifetimes)]
     fn try_find_remote_inner<'a>(
         &self,
         name_or_url: impl Into<&'a BStr>,
         rewrite_urls: bool,
     ) -> Option<Result<Remote<'_>, find::Error>> {
-        fn config_spec<T: config::tree::keys::Validate>(
```

Comment:
It's a bit unfortunate that inner functions aren't supported by `momo` presently so they have to be moved out.

In this particular case, I wonder if private methods should be `momo`'d at all - if they are only called from `momo`'d functions and methods, only one instance would exist anyway.

Lastly, if it is an internal method, I would probably just use concrete types. `momo` is only useful for the `public` API where we want the convenience for the caller.

Reply:
Could have a try to see if I can fix this, though I feel that I need to change the way momo works.

Reply:
I've removed `momo` on `try_find_remote_inner` and instead put it over `try_find_remote` and `try_find_remote_without_url_rewrite`

Reply:
I've removed use of `momo` and instead manually de-`momo` them since `try_find_remote_inner` is it's too trivial to use `momo`.